src/alignment_utils.py

```diff
@@ -0,0 +1,84 @@
+import os
+import csv
+import re
+
+ALLOWED_EXTENSIONS = ['jpg', 'jpeg', 'tif']
+
+
+def align_by_ref(time_ref, target_dir, ref_seq):
+    """Aligns timestamps of the files in the given directory.
+       Writes transformation meta information to _transformation_metainf.csv file.
+
+        Args:
+            time_ref: Time reference file with timestamps from another source
+            target_dir: Target directory that contains files (e.g. images) with timestamps as filenames
+            ref_seq: Sequence from time reference file that matches the time of the first image
+    """
+    with open(time_ref, 'r') as time_ref_file:
+        values = time_ref_file.readlines()[ref_seq - 1].split(',')
+        seq = int(values[0])
+        ref_timestamp = int(values[1])
+        # get list of filenames with timestamps
+        filename_timestamps = map(
+            lambda x: int(x.split('.')[0]),
+            filter(
+                lambda x: x.split('.')[1] in ALLOWED_EXTENSIONS,
+                os.listdir(target_dir)
+            )
+        )
+        filename_timestamps.sort()
+        extension = os.listdir(target_dir)[0].split('.')[1]
```

Comment:
I would use this construction https://docs.python.org/3/library/os.path.html#os.path.splitext

It is safer for the paths like `/path/x.y/something.jpg`.